File: src/main/java/com/nurflugel/dependencyvisualizer/enums/Color.java
package com.nurflugel.dependencyvisualizer.enums;

/**  */
public enum Color {
  yellow,
  green,
  black,
  red,
  blue;

  /** Get the next color in the enum. */
  public static Color next(Color color) {
    int     ordinal = color.ordinal();
    Color[] values  = values();

    return values[(ordinal + 1) % values.length];
  }

  /** Get the next color in the enum. */
  public static Color previous(Color color) {
    int     ordinal       = color.ordinal();
    Color[] colors        = values();
    int     previousIndex = ordinal - 1;

    if (previousIndex < 0) { previousIndex = colors.length - 1; }

    return colors[previousIndex];
  }

  public static Color first() { return values()[0]; }

  public static Color last() { return values()[values().length - 1]; }
}

File: src/main/kotlin/com/nurflugel/dependencyvisualizer/enums/Color.kt
package com.nurflugel.dependencyvisualizer.enums

/**   */
enum class Color {
    yellow,
    green,
    black,
    red,
    blue;

    companion object {
        /** Get the next color in the enum.  */
        fun next(color: Color): Color {
            val ordinal = color.ordinal
            return entries[(ordinal + 1) % entries.size]
        }

        /** Get the previous color in the enum.  */
        fun previous(color: Color): Color {
            val ordinal = color.ordinal
            var previousIndex = ordinal - 1
            if (previousIndex < 0) {
                previousIndex = entries.size - 1
            }

            return entries[previousIndex]
        }
    }
}
